### PR TITLE
Add Runtime Type Validation for Tool Results

### LINKED ISSUE #80: Inspector UI Crash when CallToolResult content is not an array
I've followed the weather example from the [docs](https://modelcontextprotocol.io/docs/first-server/typescript), and the inspector crashes when the `get_forecast` handler returns with this error in the browser:

```sh
Uncaught TypeError: f.content.map is not a function
    at u (index-jOz_Q5JL.js:185:28299)
    at qE (index-jOz_Q5JL.js:185:30526)
    at $u (index-jOz_Q5JL.js:38:16998)
    at bc (index-jOz_Q5JL.js:40:3139)
    at Th (index-jOz_Q5JL.js:40:44737)
    at Eh (index-jOz_Q5JL.js:40:39727)
    at Lx (index-jOz_Q5JL.js:40:39655)
    at ra (index-jOz_Q5JL.js:40:39508)
    at Dc (index-jOz_Q5JL.js:40:35875)
    at _h (index-jOz_Q5JL.js:40:34826)
```

[Perhaps the fastest way to explain is a quick Loom.](https://www.loom.com/share/d3d3226fcd364d73a132aa266d04409b?sid=b7fd00b5-772a-4ca8-9fc4-6c2d84db9188)

Minimal repro: https://github.com/jacksteamdev/weather-server/tree/main

### PR DESCRIPTION
**Pull request by jacksteamdev**
Fixes #80

## Problem
The inspector crashes when tool handlers return malformed responses. Specifically, when following the weather example from the docs, the inspector throws `TypeError: f.content.map is not a function` because the response structure doesn't match expectations.

## Solution
- Replace type assertions with Zod schema validation to catch malformed responses
- Add error UI to display validation failures and help debug incorrect tool responses
- Update imports to use CallToolResultSchema for runtime type checking

## Changes
- Uses `safeParse()` to validate without throwing
- Shows both invalid data and validation errors in the UI
- Maintains dark mode compatibility

## Testing
- [ ] Follow weather example from docs to verify fix
- [ ] Test various malformed response scenarios
- [ ] Check error display formatting
- [ ] Verify dark mode styles

### CODE REVIEW

**Review by dsp-ant — APPROVED** [review]
Thank you for the improvement! This is quite sensible.

File: client/src/components/ToolsTab.tsx [hunk 1]
```diff
@@ -40,7 +40,24 @@ const ToolsTab = ({
...
+      if (!parsedResult.success) {
+        return (
+          <>
+            <h4 className="font-semibold mb-2">Invalid Tool Result:</h4>
+            <pre className="bg-gray-50 dark:bg-gray-800 dark:text-gray-100 p-4 rounded text-sm overflow-auto max-h-64">
+              {JSON.stringify(toolResult, null, 2)}
+            </pre>
+            <h4 className="font-semibold mb-2">Errors:</h4>
+            {parsedResult.error.errors.map((error) => (
+              <pre className="bg-gray-50 dark:bg-gray-800 dark:text-gray-100 p-4 rounded text-sm overflow-auto max-h-64">
```
Comment (ashwin-ant) [inline client/src/components/ToolsTab.tsx]: nit: I think this might need a `key` prop

**Review by ashwin-ant — APPROVED** [review]
Thank you! Left one small comment

File: client/src/components/ToolsTab.tsx (hunk 1 above)
Comment (ashwin-ant) [inline client/src/components/ToolsTab.tsx]:
```suggestion
            {parsedResult.error.errors.map((error, idx) => (
              <pre 
                key={idx}
                className="bg-gray-50 dark:bg-gray-800 dark:text-gray-100 p-4 rounded text-sm overflow-auto max-h-64"
              >
```

**Review by ashwin-ant — APPROVED** [review]

File: client/src/components/ToolsTab.tsx (hunk 1 above)
Comment (jacksteamdev) [reply client/src/components/ToolsTab.tsx]: @ashwin-ant Perfect! I've applied your suggestion.

**Pull request merged**